bin/compile

```diff
@@ -42,25 +44,43 @@ echo "Installing graphicsmagick $GRAPHICS_MAGICK_VERSION" | arrow
 gm_lib_path=$BUILD_DIR/vendor/graphicsmagick
 mkdir -p $gm_lib_path
 
-# vendor directories
-VENDORED_GRAPHICSMAGICK="$(mktmpdir graphicsmagick)"
-archive_download "${VENDORED_GRAPHICSMAGICK}"
+graphicsmagick_cache_dir="${CACHE_DIR}/graphicsmagick-archive"
 
-pushd $VENDORED_GRAPHICSMAGICK
-./configure
-make
-make DESTDIR=$gm_lib_path install
-popd
+# Test if cache directory exists
+if [ ! -d "$graphicsmagick_cache_dir" ]; then
+  # create vendor directories
+  mktmpdir $graphicsmagick_cache_dir
+fi
+
+# Test if the archive is present in the cache directory then download it if not.
+if [ ! -d "${graphicsmagick_cache_dir}/utilities" ]; then
+  echo "Downloading archive in $graphicsmagick_cache_dir" | arrow
+  archive_download "$graphicsmagick_cache_dir"
+fi
```

Comment:
Sorry I missed it during the previous reviews... But the cache should be invalidated if you update the GraphicsMagick version (`GRAPHICS_MAGICK_VERSION`). By reading the code I feel like it does not get invalidated if the version changes. We usually put the version in the cache folder name so that the cache gets invalidated when the version changes. Do you see what I mean?